app/Main.hs

```diff
@@ -10,8 +10,9 @@ main :: IO ()
 main = do
     terms <- pure . lines =<< getContents
     results <- pure . concat =<< mapM search terms
+    let response = responseFromParse $ parseLines $ unlines results
 
-    case parseLines $ unlines results of
+    case (withOneOccurrence . skipRailsExtras . withOneFile) response of
```

Comment:
This is kind of a weird way to format this. I think there are two established idioms:

```hs
withOneOccurrence $ skipRailsExtras $ withOneFile response -- my preference
withOneOccurrence . skipRailsExtras $ withOneFile response -- also common
```